src/System.Data.Common/src/System/Data/Common/DbConnection.cs

```diff
@@ -35,6 +35,8 @@ protected DbConnection() : base()
         /// </summary>
         protected virtual DbProviderFactory DbProviderFactory => null;
 
+        internal DbProviderFactory ProviderFactory=>DbProviderFactory;
```

Comment:
If it's important enough to add back an internal member because folks are using reflection to get at it, whether or not we "officially" support that we should have a test in place that similarly uses reflection to get it and verifies the result... otherwise it's likely in the future someone will come along and remove it. (For that reason a comment next to it would be good.)